python/sglang/bench_offline_throughput.py

```diff
@@ -0,0 +1,160 @@
+"""
+Benchmark the throughput of using the offline LLM engine.
+This script does not launch a server.
+It accepts the same arguments as bench_latency.py
+
+# Usage
+python -m sglang.bench_offline_throughput --model-path meta-llama/Meta-Llama-3-8B-Instruct --batch 1 12 14 --input-len 256 512 --output-len 32 256 --result-filename out.jsonl
+
+"""
+
+import argparse
+import dataclasses
+import itertools
+import logging
+import time
+from typing import Dict, List, Tuple
+
+import jsonlines
+
+from sglang.api import Engine as getEngine
+from sglang.srt.server import Engine
+from sglang.srt.server_args import ServerArgs
+
+
+@dataclasses.dataclass
+class BenchArgs:
+    run_name: str = "before"
+    batch_size: Tuple[int] = (1,)
+    input_len: Tuple[int] = (1024,)
+    output_len: Tuple[int] = (16,)
+    result_filename: str = ""
+    # Plotting args
+    graph_sql: str = (
+        "select run_name, batch_size, prefill_throughput from results where run_name='before'"
+    )
+    graph_filename: str = "out.png"
+
+    @staticmethod
+    def add_cli_args(parser: argparse.ArgumentParser):
+        parser.add_argument("--run-name", type=str, default=BenchArgs.run_name)
+        parser.add_argument(
+            "--batch-size", type=int, nargs="+", default=BenchArgs.batch_size
+        )
+        parser.add_argument(
+            "--input-len", type=int, nargs="+", default=BenchArgs.input_len
+        )
+        parser.add_argument(
+            "--output-len", type=int, nargs="+", default=BenchArgs.output_len
+        )
+        parser.add_argument(
+            "--result-filename", type=str, default=BenchArgs.result_filename
+        )
+        # graphing
+        parser.add_argument("--graph-sql", type=str, default=BenchArgs.graph_sql)
+        parser.add_argument(
+            "--graph-filename", type=str, default=BenchArgs.graph_filename
+        )
+
+    @classmethod
+    def from_cli_args(cls, args: argparse.Namespace):
+        # use the default value's type to case the args into correct types.
+        attrs = [(attr.name, type(attr.default)) for attr in dataclasses.fields(cls)]
+        return cls(
+            **{attr: attr_type(getattr(args, attr)) for attr, attr_type in attrs}
+        )
+
+
+def prepare_synthetic_inputs_for_throughput_test(
+    batch_size: int, input_len: int, output_len: int
+):
+    input_ids = [[1] * input_len for _ in range(batch_size)]
```

Comment:
Instead of generating synthetic data, can we reuse the data generation from `bench_serving.py`?

Specifically, support the following arguments. You can import the code from `bench_serving.py`
https://github.com/sgl-project/sglang/blob/59a5ba9be0c7d2453aeacf61ac20bad99c04ec10/python/sglang/bench_serving.py#L1189-L1198
https://github.com/sgl-project/sglang/blob/59a5ba9be0c7d2453aeacf61ac20bad99c04ec10/python/sglang/bench_serving.py#L1209-L1245


Then, we can deprecate these arguments
```
    batch_size: Tuple[int] = (1,)
    input_len: Tuple[int] = (1024,)
    output_len: Tuple[int] = (16,)
```